be/src/agent/task_worker_pool.cpp

```diff
@@ -221,20 +222,30 @@ void TaskWorkerPool::_finish_task(const TFinishTaskRequest& finish_task_request)
     // Return result to FE
     TMasterResult result;
     uint32_t try_time = 0;
+    int32_t sleep_time_second = config::sleep_one_second;
+    int32_t max_retry_times = TASK_FINISH_MAX_RETRY;
 
-    while (try_time < TASK_FINISH_MAX_RETRY) {
+    while (try_time < max_retry_times) {
         StarRocksMetrics::instance()->finish_task_requests_total.increment(1);
         AgentStatus client_status = _master_client->finish_task(finish_task_request, &result);
 
         if (client_status == STARROCKS_SUCCESS) {
-            break;
-        } else {
-            try_time += 1;
-            StarRocksMetrics::instance()->finish_task_requests_failed.increment(1);
-            LOG(WARNING) << "finish task failed " << try_time << "/" << TASK_FINISH_MAX_RETRY
-                         << ". status_code=" << result.status.status_code;
+            // This means FE alter thread pool is full, all alter finish request to FE is meaningless
+            // so that we will sleep && retry 10 times
+            if (result.status.status_code == TStatusCode::TOO_MANY_TASKS &&
+                finish_task_request.task_type == TTaskType::ALTER) {
+                max_retry_times = ALTER_FINISH_TASK_MAX_RETRY;
+                sleep_time_second = sleep_time_second * 2;
```

Comment:
it will sleep about 1024 second for the last try? Is it OK to wait so long?

Reply:
it only happened when the finish alter task queue(default 4096) full lasts more than 10 minutes. this means FE was in heavy pressure, BE alter worker can only sleep otherwise fails.

Reply:
OK I see